terracotta/server/metadata.py

```diff
@@ -72,7 +114,53 @@ def get_metadata(keys: str) -> Response:
     """
     from terracotta.handlers.metadata import metadata
 
+    columns_schema = MetadataColumnsSchema()
+    columns = columns_schema.load(request.args).get("columns")
+
     parsed_keys = [key for key in keys.split("/") if key]
-    payload = metadata(parsed_keys)
-    schema = MetadataSchema()
+
+    payload = metadata(columns, parsed_keys)
+    schema = MetadataSchema(partial=columns is not None)
+    return jsonify(schema.load(payload))
+
+
+@METADATA_API.route("/metadata", methods=["POST"])
+def get_multiple_metadata() -> Response:
+    """Get metadata for multiple datasets
+    ---
+    post:
+        summary: /metadata
+        description:
+            Retrieve metadata for multiple datasets, identified by the
+            body payload. Desired columns can be filtered using the ?columns
+            query.
+        parameters:
+          - in: query
+            schema: MetadataColumnsSchema
+          - in: body
+            schema: MultipleMetadataDatasetsSchema
+        responses:
+            200:
+                description: All metadata for given dataset
+                schema: MetadataSchema
+            400:
+                description:
+                    If the MAX_POST_METADATA_KEYS (100 by default) limit is exceeded
```

Comment:
Mentioning the default here could confuse end users (since the deployment they're looking at could have a completely different limit).

```suggestion
                    If the maximum number of requested datasets is exceeded
```